optuna/terminator/erroreval.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import abc
+
+import numpy as np
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.trial import Trial
+from optuna.trial._state import TrialState
+
+
+_CROSS_VALIDATION_SCORES_KEY = "terminator:cv_scores"
+
+
+class BaseErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(self, study: Study) -> float:
+        pass
+
+
+@experimental_class("3.2.0")
+class CrossValidationErrorEvaluator(BaseErrorEvaluator):
+    def evaluate(self, study: Study) -> float:
+        assert len(study.get_trials(states=(TrialState.COMPLETE,))) != 0
```

Comment:
(optional) This method seems like a public API in which case a `ValueError` would be preferred over an assertion in my opinion. Maybe this was discussed before, it's not a strong opinion.

Reply:
Actually, this was changed from `ValueError` to `assert`. https://github.com/optuna/optuna/pull/4405#discussion_r1121193853

Let us adopt `assert` in this PR.